datacube.py

```diff
@@ -0,0 +1,382 @@
+    - start_year (int): The starting year of the date range.
+    - end_year (int): The ending year of the date range.
+
+    Returns:
+    - datetime: A randomly generated date within the specified range.
+    """
+    start_date = datetime(start_year, 1, 1)
+    end_date = datetime(end_year, 12, 31)
+    days_between = (end_date - start_date).days
+    random_days = random.randint(0, days_between)
+    random_date = start_date + timedelta(days=random_days)
+    return random_date
+
+
+def get_week(year, month, day):
+    """
+    Get the week range (start_date/end_date) for a given date.
+
+    Parameters:
+    - year (int): The year of the date.
+    - month (int): The month of the date.
+    - day (int): The day of the date.
+
+    Returns:
+    - str: A string representing the start and end dates of the week in the format 'start_date/end_date'.
+    """
+    date = datetime(year, month, day)
+    start_of_week = date - timedelta(days=date.weekday())
+    end_of_week = start_of_week + timedelta(days=6)
+    start_date_str = start_of_week.strftime("%Y-%m-%d")
+    end_date_str = end_of_week.strftime("%Y-%m-%d")
+    return f"{start_date_str}/{end_date_str}"
+
+
+def get_conditions(year1, year2):
+    """
+    Get random conditions (date, year, month, day, cloud cover) within the specified year range.
+
+    Parameters:
+    - year1 (int): The starting year of the date range.
+    - year2 (int): The ending year of the date range.
+
+    Returns:
+    - tuple: A tuple containing date, year, month, day, and a constant cloud cover value.
+    """
+    date = random_date(year1, year2)
+    YEAR = date.year
+    MONTH = date.month
+    DAY = date.day
+    CLOUD = 50
+    return date, YEAR, MONTH, DAY, CLOUD
+
+
+def search_sentinel2(week, aoi):
+    """
+    Search for Sentinel-2 items within a given week and area of interest (AOI).
+
+    Parameters:
+    - week (str): The week in the format 'start_date/end_date'.
+    - aoi (shapely.geometry.base.BaseGeometry): Geometry object for an Area of Interest (AOI).
+
+    Returns:
+    - tuple: A tuple containing the STAC catalog, Sentinel-2 items, Sentinel-2 GeoDataFrame, and the bounding box (BBOX).
+    """
+
+    CENTROID = aoi.centroid
+    BBOX = aoi.bounds
+
+    catalog = pystac_client.Client.open(STAC_API, modifier=pc.sign_inplace)
+
+    search: pystac_client.item_search.ItemSearch = catalog.search(
+        filter_lang="cql2-json",
+        filter={
+            "op": "and",
+            "args": [
+                {
+                    "op": "s_intersects",
+                    "args": [{"property": "geometry"}, CENTROID.__geo_interface__],
+                },
+                {"op": "anyinteracts", "args": [{"property": "datetime"}, week]},
+                {"op": "=", "args": [{"property": "collection"}, "sentinel-2-l2a"]},
+                {"op": "<=", "args": [{"property": "eo:cloud_cover"}, 50]},
+                {"op": "<=", "args": [{"property": "s2:nodata_pixel_percentage"}, 20]},
```

Comment:
The `eo:cloud_cover` and `s2:nodata_pixel_percentage` parameters should be made configurable perhaps?

Reply:
Yes, but maybe we can make a separate PR for this where we run at scale for all configurable options?

Reply:
It's an easy change to make so I this in [fd23e2c](https://github.com/Clay-foundation/model/commit/fd23e2cad093d6eea87e979bbb5f7d20c88b540e)